shenyu-client/shenyu-client-dubbo/shenyu-client-apache-dubbo/src/main/java/org/apache/shenyu/client/apache/dubbo/ApacheDubboServiceBeanListener.java

```diff
@@ -118,18 +125,47 @@ private void handler(final ServiceBean<?> serviceBean) {
         if (AopUtils.isAopProxy(refProxy)) {
             clazz = AopUtils.getTargetClass(refProxy);
         }
+        final ShenyuDubboClient beanShenyuClient = AnnotationUtils.findAnnotation(clazz, ShenyuDubboClient.class);
+        final String superPath = buildApiSuperPath(clazz);
+        // Compatible with previous versions
+        if (superPath.contains("*")) {
+            Method[] methods = ReflectionUtils.getDeclaredMethods(clazz);
+            for (Method method : methods) {
+                publisher.publishEvent(buildMetaDataDTO(serviceBean, beanShenyuClient, method, superPath));
+            }
+            return;
+        }
         Method[] methods = ReflectionUtils.getUniqueDeclaredMethods(clazz);
         for (Method method : methods) {
-            ShenyuDubboClient shenyuDubboClient = method.getAnnotation(ShenyuDubboClient.class);
-            if (Objects.nonNull(shenyuDubboClient)) {
-                publisher.publishEvent(buildMetaDataDTO(serviceBean, shenyuDubboClient, method));
+            ShenyuDubboClient methodShenyuClient = method.getAnnotation(ShenyuDubboClient.class);
+            if (Objects.nonNull(methodShenyuClient)) {
+                publisher.publishEvent(buildMetaDataDTO(serviceBean, methodShenyuClient, method, superPath));
+            }
+        }
+    }
+
+    private String buildApiSuperPath(@NonNull final Class<?> clazz) {
+        ShenyuDubboClient shenyuDubboClient = AnnotationUtils.findAnnotation(clazz, ShenyuDubboClient.class);
```

Comment:
Why call `AnnotationUtils.findAnnotation(clazz, ShenyuDubboClient.class);` twice ?

Reply:
Once to get the path and once to build metadatadto

Reply:
Can we get it only once?

Reply:
Of course, I have modified it